web_src/js/components/RepoActionView.vue

```diff
@@ -425,6 +431,23 @@ const sfc = {
       } else {
         actionBodyEl.append(fullScreenEl);
       }
+    },
+    async hashChangeListener() {
+      const selectedLogStep = window.location.hash;
+      const [_, step, _line] = selectedLogStep.split('-');
+      if (!this.currentJobStepsStates[step]) return;
+      if (!this.currentJobStepsStates[step].expanded && this.currentJobStepsStates[step].cursor === null) {
+        this.currentJobStepsStates[step].expanded = true;
+        // need to await for load job if the step log is loaded for the first time
+        // so logline can be selected by querySelector
+        await this.loadJob();
+      }
+      const logline = this.$refs.steps.querySelector(`${selectedLogStep}`);
+      if (!logline) return;
+      const logSummary = this.$refs.steps.querySelector(`.job-step-section:nth-of-type(${parseInt(step) + 1}) > .job-step-summary`);
+      const offset = logSummary.offsetHeight + document.querySelector('.job-info-header').offsetHeight;
```

Comment:
Does it guarantee that  `logSummary` always exist?

Reply:
If `logline` exists, then `logSummary` should exist because `logSummary` and `job-step-logs` are in the same `v-for` loop, and `logline` is child of `job-step-logs`. But might need to confirm this again if in the future `beginLogGroup` method is used.

Reply:
Update: Don't need `logSummary ` any more after https://github.com/go-gitea/gitea/pull/25532/commits/987f29cda9e80a48701248182479bbc2449f0f80